internal/discover/symlinks.go

```diff
@@ -60,7 +71,14 @@ func (d *additionalSymlinks) Hooks() ([]Hook, error) {
 		}
 		processedPaths[mount.Path] = true
 
-		for _, link := range d.getLinksForMount(mount.Path) {
+		linksForMount := d.getLinksForMount(mount.Path)
+		soSymlinks, err := d.getDotSoSymlinks(mount.HostPath)
+		if err != nil {
+			soSymlinks = nil
+		}
```

Comment:
Why would soSymlinks be anything other than `nil` if this function returns an error? By conventions we shouldn't return a value if there is an error, it should be `nil` already.

Reply:
You're right it will always be nil in our implementation. I was being overly cautious.